docs/sequence/PISPLinkingApi.puml

```diff
@@ -0,0 +1,577 @@
+@startuml
+
+title PISP Linking
+
+participant "PISP Backend" as PISP
+box "PISP tp-scheme-adapter"
+  participant "outbound-server" as PISP_TP_OUT
+  participant "PISPLinkingModel" as PISP_LM
+  participant "inbound-server" as PISP_TP_IN
+end box
+box "Mojaloop"
+    participant Switch
+    participant "Auth Service" as AUTH
+    participant "Account Lookup Service" as ALS
+end box
+box "DFSP tp-scheme-adapter"
+  participant "inbound-server" as DFSP_TP_IN
+  participant "DFSPLinkingModel" as DFSP_LM
+end box
+participant DFSPAuthorizeSimulator
+
+== Request Consent - Web ==
+autonumber 1 "<b>REQUEST-CONSENT-#</b>"
+rnote right of PISP
+PISP presents accounts to user. User selects one or more accounts to link.
+end note
+PISP -> PISP_TP_OUT: POST /linking/request-consent
+rnote right of PISP #LightBlue
+POST /linking/request-consent
+{
+  "accounts": [
+    { accountNickname: "XXXXXXnt", id: "dfspa.username.1234", currency: "ZAR" },
+    { accountNickname: "SpeXXXXXXXXnt", id: "dfspa.username.5678", currency: "USD" }
+  ],
+  callbackURI: 'pisp-app://callback'
+}
+end note
+
+
+activate PISP
+activate PISP_TP_OUT
+
+PISP_TP_OUT -> PISP_TP_OUT: const model = await create()
+rnote right of PISP_TP_OUT: state: start
+PISP_TP_OUT -> PISP_LM: model.requestConsent()
+
+activate PISP_LM
+
+PISP_LM -> PISP_LM: ThirdpartyRequests.postConsentRequests()
+PISP_LM -> Switch: POST /consentRequests
+rnote right of PISP_LM #LightBlue
+POST /consentRequests
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+{
+  // consentRequestId will be the uuid created
+  // at the PRELINKING-2
+  consentRequestId: 6789
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  // model will add `authChannels`
+  authChannels: ["WEB", "OTP"],
+  callbackURI: 'pisp-app://callback...'
+}
+end note
+
+activate Switch
+Switch --> PISP_LM: 202 Accepted
+deactivate PISP_LM
```

Comment:
drop `deactivate PISP_LM` to express that model is listening on PUT `/consentReqeusts/8789`